pkg/kudoctl/util/repo/repo_operator.go

```diff
@@ -79,21 +79,43 @@ func (r *Client) DownloadIndexFile() (*IndexFile, error) {
 }
 
 // getPackageReaderByFullPackageName downloads the tgz file from the remote repository and unmarshals it to the package CRDs
-func (r *Client) getPackageReaderByFullPackageName(fullPackageName string) (io.Reader, error) {
-	var fileURL string
-	parsedURL, err := url.Parse(r.Config.URL)
+// this will cycle through urls for the
+func (r *Client) getPackageReaderByFullPackageName(pkg *PackageVersion) (io.Reader, error) {
+
+	pkgURL, err := repositoryBackedPkgURL(r.Config.URL, pkg)
 	if err != nil {
-		return nil, errors.Wrap(err, "parsing config url")
+		return nil, err
 	}
-	parsedURL.Path = fmt.Sprintf("%s/%s.tgz", parsedURL.Path, fullPackageName)
+	urls := append(pkg.URLs, pkgURL)
 
-	fileURL = parsedURL.String()
-	clog.V(4).Printf("using url: %v", fileURL)
+	var pkgErr error
+	for _, url := range urls {
+		r, err := r.getPackageReaderByURL(url)
+		if err == nil {
+			return r, nil
+		}
+		pkgErr = fmt.Errorf("unable to read package %w", err)
+		clog.Errorf("failure against url: %v  %v", url, pkgErr)
+	}
+	clog.Printf("Giving up with err %v", pkgErr)
+	return nil, pkgErr
+}
 
-	return r.getPackageReaderByURL(fileURL)
+func repositoryBackedPkgURL(repoURL string, pkg *PackageVersion) (string, error) {
+	packageName := pkg.Name + "-" + pkg.Version
```

Comment:
I am very confused here, we have a type called PackageVersion, that contains both name and version. I think we need a better name for that type plus composing this package name I think should be a method on that type, since we already have that type.